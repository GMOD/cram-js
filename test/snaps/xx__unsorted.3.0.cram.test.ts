import { testFile } from '../testUtil'
testFile('xx#unsorted.3.0.cram')
