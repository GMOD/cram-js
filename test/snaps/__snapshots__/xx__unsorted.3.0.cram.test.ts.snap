// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole xx#unsorted.3.0.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 4272341097,
      "landmarks": [
        0,
        130,
      ],
      "length": 204,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 4272341097,
        "landmarks": [
          0,
          130,
        ],
        "length": 204,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "yy",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "yy",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              147,
            ],
          },
          "parametersBytes": 5,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -20,
            ],
          },
          "parametersBytes": 8,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 267,
      "_size": 18,
      "alignmentSpan": 10,
      "crc32": 2128734450,
      "landmarks": [
        181,
      ],
      "length": 292,
      "numBases": 10,
      "numBlocks": 5,
      "numLandmarks": 1,
      "numRecords": 1,
      "recordCounter": 0,
      "refSeqId": 1,
      "refSeqStart": 11,
    },
    "data": [
      {
        "_endPosition": 267,
        "_size": 18,
        "alignmentSpan": 10,
        "crc32": 2128734450,
        "landmarks": [
          181,
        ],
        "length": 292,
        "numBases": 10,
        "numBlocks": 5,
        "numLandmarks": 1,
        "numRecords": 1,
        "recordCounter": 0,
        "refSeqId": 1,
        "refSeqStart": 11,
      },
      {
        "features": [
          {
            "alignmentStart": 11,
            "cramFlags": 3,
            "flags": 147,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 1,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "b1",
            "sequenceId": 1,
            "tags": {},
            "templateSize": -20,
            "uniqueId": 454,
          },
        ],
        "header": {
          "_endPosition": 487,
          "_size": 34,
          "compressedSize": 30,
          "compressionMethod": "raw",
          "content": Uint8Array [
            1,
            11,
            10,
            1,
            0,
            3,
            2,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            130,
            12,
            146,
            46,
            18,
            207,
            168,
            96,
            235,
            24,
            29,
            98,
            105,
            199,
            126,
            99,
          ],
          "contentId": 0,
          "contentPosition": 453,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2667260533,
          "parsedContent": {
            "_endPosition": 297,
            "_size": 30,
            "contentIds": [
              11,
              12,
            ],
            "md5": [
              130,
              12,
              146,
              46,
              18,
              207,
              168,
              96,
              235,
              24,
              29,
              98,
              105,
              199,
              126,
              99,
            ],
            "numBlocks": 3,
            "numContentIds": 2,
            "numRecords": 1,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 1,
            "refSeqSpan": 10,
            "refSeqStart": 11,
          },
          "uncompressedSize": 30,
        },
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": false,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 6,
          "parameters": {
            "length": 4,
            "offset": 0,
          },
          "parametersBytes": 2,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              99,
              147,
            ],
          },
          "parametersBytes": 7,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              1,
              5,
            ],
          },
          "parametersBytes": 6,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 577,
      "_size": 18,
      "alignmentSpan": 20,
      "crc32": 457948287,
      "landmarks": [
        154,
      ],
      "length": 290,
      "numBases": 20,
      "numBlocks": 5,
      "numLandmarks": 1,
      "numRecords": 2,
      "recordCounter": 1,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 577,
        "_size": 18,
        "alignmentSpan": 20,
        "crc32": 457948287,
        "landmarks": [
          154,
        ],
        "length": 290,
        "numBases": 20,
        "numBlocks": 5,
        "numLandmarks": 1,
        "numRecords": 2,
        "recordCounter": 1,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 11,
            "cramFlags": 5,
            "flags": 147,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "a1",
              "sequenceId": 0,
              "uniqueId": 739,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 738,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 1,
            "flags": 99,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "a1",
              "sequenceId": 0,
              "uniqueId": 738,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 739,
          },
        ],
        "header": {
          "_endPosition": 770,
          "_size": 34,
          "compressedSize": 30,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            20,
            2,
            1,
            3,
            2,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            187,
            244,
            222,
            109,
            132,
            151,
            161,
            25,
            221,
            166,
            224,
            116,
            82,
            22,
            67,
            220,
          ],
          "contentId": 0,
          "contentPosition": 736,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 1127133598,
          "parsedContent": {
            "_endPosition": 607,
            "_size": 30,
            "contentIds": [
              11,
              12,
            ],
            "md5": [
              187,
              244,
              222,
              109,
              132,
              151,
              161,
              25,
              221,
              166,
              224,
              116,
              82,
              22,
              67,
              220,
            ],
            "numBlocks": 3,
            "numContentIds": 2,
            "numRecords": 2,
            "recordCounter": 1,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 20,
            "refSeqStart": 1,
          },
          "uncompressedSize": 30,
        },
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": false,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 6,
          "parameters": {
            "length": 4,
            "offset": 0,
          },
          "parametersBytes": 2,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              99,
              147,
            ],
          },
          "parametersBytes": 7,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              2,
            ],
            "numCodes": 3,
            "numLengths": 3,
            "symbols": [
              5,
              1,
              3,
            ],
          },
          "parametersBytes": 8,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              11,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              1,
            ],
          },
          "parametersBytes": 6,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              20,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 889,
      "_size": 22,
      "alignmentSpan": 0,
      "crc32": 2721024782,
      "landmarks": [
        191,
      ],
      "length": 335,
      "numBases": 30,
      "numBlocks": 5,
      "numLandmarks": 1,
      "numRecords": 3,
      "recordCounter": 3,
      "refSeqId": -2,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 889,
        "_size": 22,
        "alignmentSpan": 0,
        "crc32": 2721024782,
        "landmarks": [
          191,
        ],
        "length": 335,
        "numBases": 30,
        "numBlocks": 5,
        "numLandmarks": 1,
        "numRecords": 3,
        "recordCounter": 3,
        "refSeqId": -2,
        "refSeqStart": 0,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 99,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "flags": 1,
              "readName": undefined,
              "sequenceId": 1,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "b1",
            "sequenceId": 1,
            "tags": {},
            "templateSize": 20,
            "uniqueId": 1089,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "c1",
              "sequenceId": 0,
              "uniqueId": 1091,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "c1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 1090,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "c1",
              "sequenceId": 0,
              "uniqueId": 1090,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "c1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 1091,
          },
        ],
        "header": {
          "_endPosition": 1123,
          "_size": 38,
          "compressedSize": 34,
          "compressionMethod": "raw",
          "content": Uint8Array [
            255,
            255,
            255,
            255,
            14,
            0,
            0,
            3,
            3,
            3,
            2,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ],
          "contentId": 0,
          "contentPosition": 1085,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 975208009,
          "parsedContent": {
            "_endPosition": 923,
            "_size": 34,
            "contentIds": [
              11,
              12,
            ],
            "md5": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "numBlocks": 3,
            "numContentIds": 2,
            "numRecords": 3,
            "recordCounter": 3,
            "refBaseBlockId": -1,
            "refSeqId": -2,
            "refSeqSpan": 0,
            "refSeqStart": 0,
          },
          "uncompressedSize": 34,
        },
      },
    ],
  },
]
`;
